app/policies/task_policy.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+class TaskPolicy < ApplicationPolicy
+  def task
+    @record
+  end
+
+  def show?
+    if @user.present?
+      admin? || task.access_level_public? || task.in_same_group?(@user) || task.author?(@user)
+    else
+      task.access_level_public?
+    end
+  end
+
+  %i[import_uuid_check? import_external?].each do |action|
+    define_method(action) { true } # allow unauthenticated access for API routes
```

Comment:
Since we are authenticated with a `current_user` in the action (and by moving that to a `before_filter` that is performed before `#only_authorize_action`), we should also be able to switch that to `everyone` and get rid of the comment.

Reply:
That would throw an error in case an API request without a user comes in. This would again not return a JSON but instead the pundit NotAuthorizedError. I dont know how we would get around that.

Reply:
Same as above. In the `ApplicationController`, we catch any occurrence of the `Pundit::NotAuthorizedError`:

https://github.com/Mathis-Z/codeharbor/blob/499405041763852022acc6ac10718eaaa4e8b490/app/controllers/application_controller.rb#L17

Then, the `render_error` method is called:

https://github.com/Mathis-Z/codeharbor/blob/499405041763852022acc6ac10718eaaa4e8b490/app/controllers/application_controller.rb#L61-L63

Which, in turn, either redirects for any format except JSON or renders a JSON:

https://github.com/Mathis-Z/codeharbor/blob/499405041763852022acc6ac10718eaaa4e8b490/app/controllers/application_controller.rb#L73-L89